lib/chef/knife/ssh.rb

```diff
@@ -103,8 +103,14 @@ class Ssh < Knife
         :boolean => true,
         :default => true
 
+      option :exit_on_error,
+        :short => '-e',
+        :long => '--exit-on-error',
+        :description => "Immediately exit if an error is encountered",
+        :proc => Proc.new { Chef::Config[:knife][:on_error] = :raise }
+
       def session
-        config[:on_error] ||= :skip
+        config[:on_error] ||= Chef::Config[:knife][:on_error] || :skip
```

Comment:
`config[:on_error]` and `Chef::Config[:knife][:on_error]` are already merged by knife in `Chef::Knife#merge_configs`.

I think all you probably need to do here is have the mixlib-cli option be ':on_error' and then set it to `:raise` or `:skip` appropriately and then it'll get merged correctly in the same precedence and everything with the other CLI options.